sklearn/metrics/scorer.py

```diff
@@ -279,6 +289,9 @@ def make_scorer(score_func, greater_is_better=True, needs_proba=False,
         For example ``average_precision`` or the area under the roc curve
         can not be computed using discrete predictions alone.
 
+    name : str, default=None
+        The name for the scorer. If None, the score_func.__name__ is used.
```

Comment:
I think we lose very little by requiring that the user be explicit. I've already suggested that magic such as `__name__` is very brittle. Let's just default to `''` or `'score'` or something magic-free.